packages/flutter/test/widgets/draggable_scrollable_sheet_test.dart

```diff
@@ -1746,4 +1746,30 @@ void main() {
       areCreateAndDispose,
     );
   });
+
+  testWidgets('DraggableScrollableSheet respects shouldCloseOnMinExtent', (WidgetTester tester) async {
+    final DraggableScrollableController controller = DraggableScrollableController();
+    DraggableScrollableNotification? receivedNotification;
+
+    Future<void> pumpWidgetAndFling() async {
+      await tester.pumpWidget(boilerplateWidget(
+        null,
+        controller: controller,
+        shouldCloseOnMinExtent: false,
+        onDraggableScrollableNotification: (DraggableScrollableNotification notification) {
+          receivedNotification = notification;
+          return false;
+        },
+      ));
+
+      await tester.flingFrom(const Offset(0, 325), const Offset(0, -325), 200);
+      await tester.pumpAndSettle();
+    }
+
+    await pumpWidgetAndFling();
+    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);
+    controller.jumpTo(0.5);
+    await pumpWidgetAndFling();
+    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);
+  });
```

Comment:
Hadn't noticed this till now, but it looks like `DraggableScrollableController` is a `ChangeNotifier`, so we'll want to `dispose` of it to prevent memory leaks.

```suggestion
    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);
    controller.dispose();
  });
```